pkg/api/component_report.go

```diff
@@ -0,0 +1,647 @@
+package api
+
+import (
+	"context"
+	"regexp"
+	"sort"
+	"strings"
+	"sync"
+	"time"
+
+	"cloud.google.com/go/bigquery"
+	fischer "github.com/glycerine/golang-fisher-exact"
+	apitype "github.com/openshift/sippy/pkg/apis/api"
+	"github.com/openshift/sippy/pkg/util/sets"
+	"github.com/pkg/errors"
+	log "github.com/sirupsen/logrus"
+	"google.golang.org/api/iterator"
+)
+
+func getSingleColumnResultToSlice(query *bigquery.Query, names *[]string) error {
+	it, err := query.Read(context.TODO())
+	if err != nil {
+		log.WithError(err).Error("error querying test status from bigquery")
+		return err
+	}
+
+	for {
+		row := struct{ Name string }{}
+		err := it.Next(&row)
+		if err == iterator.Done {
+			break
+		}
+		if err != nil {
+			log.WithError(err).Error("error parsing component from bigquery")
+			return err
+		}
+		*names = append(*names, row.Name)
+	}
+	return nil
+}
+
+var (
+	cacheLock     = new(sync.RWMutex)
+	cacheVariants *apitype.ComponentReportTestVariants
+)
+
+func GetComponentTestVariantsFromBigQuery(client *bigquery.Client) (apitype.ComponentReportTestVariants, []error) {
+	errs := []error{}
+	result := apitype.ComponentReportTestVariants{}
+	cacheLock.RLock()
+	if cacheVariants != nil {
+		result = *cacheVariants
+		cacheLock.RUnlock()
+		return result, errs
+	}
+	cacheLock.RUnlock()
```

Comment:
Yes, 7m total

Reply:
If every one of them takes around 1s, it could still be long. So caching is needed. 

@jupierce This is about BigQuery setting tweak to speed up queries. Is the setting you mentioned before done from the server side? Or it can be configured from the client side? Maybe we have a golang equivalent I can consider.